### ISSUE TITLE
Typed dictionary causes godot-cpp compilation error because of not being unable to parse and resolve `typeddictionary::` into correct format

### ISSUE THREAD

**Issue by Lazy-Rabbit-2001** [godotengine/godot#100137]
### Tested versions

4.4 dev6

### System information

Windows 11

### Issue description

See: https://github.com/godotengine/godot/actions/runs/12211086323/job/34068068226?pr=100135
Using `TypedDictionary<>()` will lead to failture of godot-cpp compilation. Needs to be fixed as soon as possible.

### Steps to reproduce

Using a `TypedDictionary` in source code and push it to your forked repo, and then run the action on your fork.

### Minimal reproduction project (MRP)

Reproduction = source code. See **Steps to reproduce**

**Comment by Zylann** [comment]
I'm not sure this is a Godot issue? The format looks ok, `typeddictionary::Key;Value`, are you using the latest GodotCpp? It should support them since this PR https://github.com/godotengine/godot-cpp/pull/1162, otherwise that would be a GodotCpp issue.

**Comment by AThousandShips** [comment]
Could be because the CI checks against `4.3`

**Comment by dsnopek** [comment]
> Could be because the CI checks against `4.3`

Yeah, we use the previous version of godot-cpp in the Godot tests to avoid circular dependencies between Godot and godot-cpp changes.

But updates to `extension_api.json` are tricky, if this is the issue, we may need to either (1) use a particular commit from `master` in the Godot CI, or (2) merge something to the `4.3` branch of godot-cpp that allows it to not generate bad code when there's a typed dictionary in `extension_api.json` (it could just use `Dictionary`?)

I feel like we should have had this same problem with typed arrays, but I don't recall if we had to do anything to for that?

**Comment by AThousandShips** [comment]
Typed arrays were introduced a long time ago, not sure we had any checks for extensions back then
